### PR TITLE
Move requirements to a folder

### PR DESCRIPTION
**Pull request by andrey-canon**
## Description
Move requirements into a folder
update settings
add upgrade command

### CODE REVIEW

File: eox_core/settings/test.py [hunk 1]
```diff
@@ -58,3 +60,27 @@ def plugin_settings(settings):  # pylint: disable=function-redefined
 
 USERNAME_REGEX_PARTIAL = r'[\w .@_+-]+'
 USERNAME_PATTERN = r'(?P<username>{regex})'.format(regex=USERNAME_REGEX_PARTIAL)
+
+PROCTORING_SETTINGS = {
```
Comment (morenol) [inline eox_core/settings/test.py:64]: Why is this added in this PR?

File: requirements/base.in [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+celery
+Django
+djangorestframework
+django-filter
+django-oauth2-provider
+django-waffle
+edx-proctoring
+edx-opaque-keys
+factory-boy
```
Comment (morenol) [inline requirements/base.in:13]: Is this a test package?

File: requirements/base.in [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+-r github.in
+
+celery
+Django
+djangorestframework
+django-filter
+django-oauth2-provider
+django-waffle
+edx-proctoring
+edx-opaque-keys
```
Comment (morenol) [inline requirements/base.in:12]: setup.py has some of these dependencies, should we remove them from there? or change setup.py to use this as dependencies?

In case of the second, I think that we need to change this with edx-opaque-keys[django] in order to prevent error when this package is installed using `pip install eox-core`

File: eox_core/settings/test.py (hunk 1 above)
Comment (andrey-canon) [reply eox_core/settings/test.py:64]: the version 1.5.7 tries to import  signals and signals.py tries to import the constants https://github.com/edx/edx-proctoring/blob/1.5.7/edx_proctoring/apps.py#L117,  the previous version was 1.4.0, but that version is not the platform version.

**Comment by andrey-canon** [comment]
> I added a few comments, please add an entry in the changelog.

@morenol  does it make any sense to  add refactor or quality changes  in the changelog ?, I mean, this pr doesn't modify, add or fix anything, so if someone fixes a typo, a new entry will be required?, imo the changelog only should contain functionality changes, btw I have already added  the entry.

File: Makefile [hunk 4]
```diff
@@ -35,3 +35,8 @@ python-quality-test:
 
 javascript-quality-test:
 	./node_modules/.bin/eslint ./eox_core/**/*.js*
+
```
Comment (morenol) [inline Makefile:38]:
```suggestion

upgrade: export CUSTOM_COMPILE_COMMAND=make upgrade
```
In order to get the generated .txt files have this instead of the `pip-compile` complete command.

File: Makefile [hunk 5]
```diff
@@ -35,3 +35,8 @@ python-quality-test:
...
+upgrade: ## update the requirements/*.txt files with the latest packages satisfying requirements/*.in
+	pip install -q pip-tools
```
Comment (morenol) [inline Makefile:40]: I think that we should create a pip-tools.in and a pip-tools.txt files with this package

File: Makefile [hunk 6]
```diff
@@ -35,3 +35,8 @@ python-quality-test:
 
 javascript-quality-test:
 	./node_modules/.bin/eslint ./eox_core/**/*.js*
+
+upgrade: ## update the requirements/*.txt files with the latest packages satisfying requirements/*.in
```
Comment (morenol) [inline Makefile:39]: I think that we should roon pip-compile with this flag:  `--upgrade`

**Comment by morenol** [comment]
> @morenol does it make any sense to add refactor or quality changes in the changelog ?, I mean, this pr doesn't modify, add or fix anything, so if someone fixes a typo, a new entry will be required?, .

You are right, but maybe the change related to the PROCTORING_SETTINGS is worth

File: Makefile (hunk 4 above)
Comment (andrey-canon) [reply Makefile:42]: Updated to the last version on the cookiecutter

**Comment by andrey-canon** [comment]
> ht, but maybe the change related to the PROCTORING_SETTINGS is worth

@morenol  something like `Add proctoring test settings since this had the wrong proctoring version` ???

File: requirements/base.in [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+-r github.in
+
+celery
+Django
+djangorestframework
+django-filter
+django-oauth2-provider
+django-waffle
+edx-proctoring
+edx-opaque-keys[django]
```
Comment (Squirrel18) [inline requirements/base.in:12]: Where's the six package? I think we should explicitly add it. @andrey-canon
